### PR TITLE
I2c

### PR DESCRIPTION
**Pull request by rs2thoma**

### CODE REVIEW

**Comment by amc1234** [comment]
Did you test this with a i2c device?

File: i2c.py [hunk 1]
```diff
@@ -0,0 +1,14 @@
+import smbus
+
+class I2C():
+    # Channel: I2C channel; 0 or 1
+    def __init__(self, channel):
+        self.bus = smbus.SMBus(channel)
+
+    # Addr: Address, Reg: Register, val: array of 32 bit values to write
```
Comment (tiger-hao) [inline i2c.py:8]: just curious, does an address contain registers?

**Review by tiger-hao — APPROVED** [review]

**Comment by tiger-hao** [comment]
code looks good

**Review by amc1234 — APPROVED** [review]

File: i2c.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    def write(self, addr, reg, val):
+        self.bus.write_block_data(addr, reg, val)
+
+    # Writes a byte directly without sending the register value; used by mux
+    def write_byte(self, addr, val)
```
Comment (tiger-hao) [inline i2c.py]: missing a colon

File: i2c.py [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+        self.bus.write_byte(addr, val)
+
+    # Reads a byte directly without sending the register value; used by mux
+    def read_byte(self, addr, val)
```
Comment (tiger-hao) [inline i2c.py]: missing a colon

**Review by tiger-hao — APPROVED** [review]

**Pull request merged**